lasio/las.py

```diff
@@ -430,17 +434,24 @@ def read(
                     if isinstance(dtypes, dict):
                         dtypes = [dtypes.get(c.mnemonic, float) for c in self.curves]
 
-                    # Notes see 2d9e43c3 and e960998f for 'try' background
+                    # ----------------------------------------------------------------------
+                    # Notes
+                    # see 2d9e43c3 and e960998f for 'try' background
+                    # 2023-03-03: dcs:
+                    #  With the addtion of "Exception" to "except Exception" the
+                    # "except KeybboardInterrupt" shouldn't be needed because
+                    # "except Exception" won't catch the KeyboardInterrupt exception
+                    # .. verify before removing, by Cntrl-C in the middle of loading a big
+                    # las file..
+                    # ----------------------------------------------------------------------
```

Comment:
I don't understand - is this retaining the same behaviour? Why do we need to remove KeyboardInterrupt?

Reply:
It is a good question.  I may have this wrong, so review my thinking below and the code can be adjusted if/as needed.

The current verison on main, uses a bare (ie no specified exception) `except:`,   A bare except will  catch the `KeyboardInterupt`.  To prevent this the bare except is preceded by the `except KeyboardInterrupt:` that does a generic `raise` that, I think, propagates up and exits the script.

In this changed version, the `except:` has been changed to `except Exception:`.  The `Exception` exception doesn't catch `KeyboardInterrupt`, so a  `KeyboardInterrupt` event would by naturally propagate up and exit the script, like it would elsewhere.   Flake8 considers `except Exception:` better than `except:`. because the exception can be captured and reported on.  In this case, I didn't implement `except Exception as err:` to log `err` because the code is immediately raising `exceptions.LASDataError()`.  However, I kind of think the `err` should be passed on to the exceptions.LASDataError() so if it is an unexpected error the information will be available to resolve it.

I did verify that `Cntrl-C `, and `KeyboardInterrupt` will exit this version, which I think is the expected behavior.  I just forgot to remove the `.. verify before removing, by Cntrl-C in the middle of loading a big las file..` comment.

Reply:
Got it - thanks, I just got a bit confused! Let's remove the comment in the next PR or another time 👍